File: solutions/excelSheetColNum/excelSheetColNum.py
#!/usr/bin/env python
# coding: utf-8

# In[10]:


class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        n = len(columnTitle)
        result = 0
        for c in columnTitle:
            result += (ord(c) - 64) * (26 ** (n-1))
            n -= 1
        return result

columnTitle = "ZY"
solver = Solution()
print(solver.titleToNumber(columnTitle))

File: solutions/excelSheetColNum/excelSheetColNum.cpp
#include <iostream>
#include <math.h>
#include <string>
#include <vector>

class Solution {
public:
    int titleToNumber(std::string columnTitle) {
        int n = columnTitle.length();
        int result = 0;
        for (char c : columnTitle) {
            result += (int(c) - 64) * pow(26, n - 1);
            n--;
        }
        return result;
    }
};

int main() {
    Solution solver;
    std::cout << solver.titleToNumber("ZY") << std::endl;
    return 0;
}
